src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -1363,19 +1363,32 @@ private static bool ShouldProceedWithRecording(List<ModelBase> models)
 
         public void DeleteModel(XmlElement modelData)
         {
+            //When there is a Redo operation, model is removed from 
+            //the workspace but the model is "not disposed" from memory.
+            //Identified this when redo operation is performed on groups
             ModelBase model = GetModelForElement(modelData);
 
             if (model is NoteModel)
-                Notes.Remove(model as NoteModel);
+            {
+                var note = model as NoteModel;
+                Notes.Remove(note);                
+                note.Dispose();
+            }
             else if (model is AnnotationModel)
-                Annotations.Remove(model as AnnotationModel);
+            {
+                var annotation = model as AnnotationModel;
+                Annotations.Remove(annotation);
+                annotation.Dispose();
+            }
             else if (model is ConnectorModel)
             {
                 var connector = model as ConnectorModel;
                 connector.Delete();
             }
             else if (model is NodeModel)
-                Nodes.Remove(model as NodeModel);
+            {
+                RemoveNode(model as NodeModel);
+            }
```

Comment:
Yes. Otherwise, Groups will never know that node is disposed.  When a node is deleted it is actually disposed. When you do Undo on that node deletion, you get the Node back. But when you do Redo, the node is removed from the workspace, but not disposed.